Glossary.md

```diff
@@ -1,19 +1,58 @@
 ## CI346 Glossary (maintained by UoB students)
 
-### Programming Languages
+1. [Programming languages](#programming-languages)
+2. [Concurrency](#concurrency)
+3. [Client-server computing](#client-server-computing)
+
+Programming Languages
+---
+
+**Object-oriented Paradigm**
+
+Object-oriented programming solves computing concepts and problems by modelling objects. Objects have states and behaviours; for example, a person has a certain hair colour (state) and this person can change the colour(behaviour). Inheritance is part of the core functionality of objects in OO languages. Classes can have subclasses, which enables the subclass to inherit the methods and fields of its superclass. The core concept of inheritance is simple, but it can be used to model complex systems.  Objects are a demonstration a core tenant of OOP, encapsulation. This mechanism is used to enclose methods and state in bundles and to restrict access to them. 
+
+Using _Java_ as the example language
```

Comment:
Not sure this is a good example under OOP. I think your above example (person, hair colour) was much better. Can we continue with that example in code? 

Generally, giving an informal introduction first, followed by a practical example is a really good way to explain something.

Reply:
Example changed to a Person, with which the hair colour changes.